Userland/Libraries/LibSemVer/SemVer.cpp

```diff
@@ -0,0 +1,305 @@
+
+    // Current version is greater than spec
+    if (compare_op == ">"sv)
+        return is_greater_than(spec_version);
+    if (compare_op == "<"sv)
+        return is_lesser_than(spec_version);
+    if (compare_op == ">="sv)
+        return is_same(spec_version) || is_greater_than(spec_version);
+    if (compare_op == "<="sv)
+        return is_same(spec_version) || is_lesser_than(spec_version);
+
+    return false;
+}
+
+ErrorOr<SemVer> from_string_view(StringView const& version, char m_number_separator)
+{
+    using namespace regex;
+    if (version.count(m_number_separator) < 2)
+        return AK::Error::from_string_view("Insufficient version separator"sv);
+
+    if (version.count('+') > 1)
+        return AK::Error::from_string_view("Build metadata must be defined once"sv);
+
+    GenericLexer lexer(version.trim_whitespace());
+    if (lexer.tell_remaining() == 0)
+        return AK::Error::from_string_view("Version string is empty"sv);
+
+    // Parse the normal version parts
+    // https://semver.org/#spec-item-2
+    auto version_part = lexer.consume_until(m_number_separator).to_uint<u64>();
+    if (!version_part.has_value())
+        return AK::Error::from_string_view("Major version is not numeric"sv);
+    auto v_major = version_part.value();
+
+    lexer.consume();
+
+    version_part = lexer.consume_until(m_number_separator).to_uint<u64>();
+    if (!version_part.has_value())
+        return AK::Error::from_string_view("Minor version is not numeric"sv);
+    auto v_minor = version_part.value();
+
+    lexer.consume();
+
+    version_part = lexer.consume_while<Function<bool(char)>>([](char ch) {
+                            return ch >= '0' && ch <= '9';
+                        })
+                       .to_uint<u64>();
+    if (!version_part.has_value())
+        return AK::Error::from_string_view("Patch version is not numeric"sv);
+    auto v_patch = version_part.value();
+
+    if (lexer.is_eof())
+        return SemVer(v_major, v_minor, v_patch, m_number_separator);
+
+    String buildmetadata;
+    Vector<String> prerelease_identifiers;
+
+    // Note: This is efficient that containsa
+    auto constexpr has_badchars = [](String const& haystack) -> bool {
+        for (const auto ch : haystack.bytes()) {
+            if (ch == ' ' || ch == '_')
+                return true;
+        }
+        return false;
+    };
+
+    switch (lexer.consume()) /* just after patch ch */
+    {
+    case '+': {
+        // Spec: https://semver.org/#spec-item-10
+        if (lexer.is_eof())
+            return AK::Error::from_string_view("Buildmeta can't be empty"sv);
+
+        buildmetadata = TRY(TRY(String::from_utf8(lexer.consume_all())).trim_ascii_whitespace());
+        if (buildmetadata.starts_with('.'))
+            return AK::Error::from_string_view("Invalid build metadata part"sv);
+        if (has_badchars(buildmetadata))
+            return AK::Error::from_string_view("Buildmeta can't contain bad characters"sv);
```

Comment:
Let's call this "build metadata" everywhere.

```suggestion
        if (lexer.is_eof())
            return AK::Error::from_string_view("Build metadata can't be empty"sv);

        buildmetadata = TRY(TRY(String::from_utf8(lexer.consume_all())).trim_ascii_whitespace());
        if (buildmetadata.starts_with('.'))
            return AK::Error::from_string_view("Invalid build metadata part"sv);
        if (has_badchars(buildmetadata))
            return AK::Error::from_string_view("Build metadata can't contain bad characters"sv);
```